packages/blade/src/components/Input/TextInput/TextInput.tsx

```diff
@@ -152,128 +154,137 @@ const isReactNative = (_textInputRef: any): _textInputRef is TextInputReactNativ
   return getPlatformType() === 'react-native';
 };
 
-export const TextInput = ({
-  label,
-  labelPosition = 'top',
-  placeholder,
-  type = 'text',
-  defaultValue,
-  name,
-  value,
-  maxCharacters,
-  onChange,
-  onFocus,
-  onBlur,
-  isDisabled,
-  necessityIndicator,
-  validationState,
-  errorText,
-  helpText,
-  successText,
-  isRequired,
-  icon,
-  prefix,
-  showClearButton,
-  onClearButtonClick,
-  isLoading,
-  suffix,
-  autoFocus,
-  keyboardReturnKeyType,
-  autoCompleteSuggestionType,
-}: TextInputProps): ReactElement => {
-  const textInputRef = React.useRef<HTMLInputElement | TextInputReactNative>(null);
-  const [shouldShowClearButton, setShouldShowClearButton] = useState(false);
+export const TextInput = React.forwardRef<NativeHTMLInputRefProps, TextInputProps>(
+  (
+    {
+      label,
+      labelPosition = 'top',
+      placeholder,
+      type = 'text',
+      defaultValue,
+      name,
+      value,
+      maxCharacters,
+      onChange,
+      onFocus,
+      onBlur,
+      isDisabled,
+      necessityIndicator,
+      validationState,
+      errorText,
+      helpText,
+      successText,
+      isRequired,
+      icon,
+      prefix,
+      showClearButton,
+      onClearButtonClick,
+      isLoading,
+      suffix,
+      autoFocus,
+      keyboardReturnKeyType,
+      autoCompleteSuggestionType,
+    },
+    ref,
+  ): ReactElement => {
+    const textInputRef = React.useRef<HTMLInputElement | TextInputReactNative>(null);
```

Comment:
we can move this directly in our custom use ref hook? so all you then need to do is 

```
const textInputRef = useMaskedRef(ref)
```
and then `useMaskedRef` will internally call `React.useRef` because otherwise we'll write this at every place and also call our custom ref hook too then pass the ref to it

Reply:
Okay. makes sense.